manifests/server/config.pp

```diff
@@ -195,7 +195,48 @@
   }
 
   if $facts['os']['family'] == 'RedHat' {
-    if $facts['os']['release']['full'] =~ /^7|^8/ or $facts['os']['name'] == 'Fedora' {
+    if $facts['os']['release']['major'] == '8' or ($facts['os']['name'] == 'Fedora' and $facts['os']['release']['major'] >= '32') {
+      # Template uses:
+      # - $::operatingsystem
+      # - $service_name
+      # - $port
+      # - $datadir
+      file {
+        default:
+            ensure => file,
+            owner  => root,
+            group  => root,
+            notify => [ Exec['restart-systemd'], Class['postgresql::server::service'] ],
+            before => Class['postgresql::server::reload'],
+
+        ;
+
+        'systemd-conf-dir':
+            ensure => directory,
+            path   => "/etc/systemd/system/${service_name}.service.d",
+        ;
+
+        'systemd-override':
+            path    => "/etc/systemd/system/${service_name}.service.d/${service_name}.conf",
+            content => template('postgresql/systemd-override.erb'),
+            require => File['systemd-conf-dir'],
+        ;
+
+        # Remove old unit file to avoid service failures
+        'old-systemd-override':
+            ensure => absent,
+            path   => "/etc/systemd/system/${service_name}.service",
+        ;
+      }
+
+      exec { 'restart-systemd':
+        command     => 'systemctl daemon-reload',
+        refreshonly => true,
+        path        => '/bin:/usr/bin:/usr/local/bin',
+      }
```

Comment:
This is only needed on Puppet < 6.1.0. Newer versions do this when needed.

Reply:
Newer versions of puppet or systemd?

Reply:
Puppet 6.1.0. See https://tickets.puppetlabs.com/browse/PUP-3483 for details.

Reply:
The exec command has been removed.